crypto/secp256k1/curve.go

```diff
@@ -231,8 +229,12 @@ func (BitCurve *BitCurve) ScalarMult(Bx, By *big.Int, scalar []byte) (*big.Int,
 
 	// Do the multiplication in C, updating point.
 	point := make([]byte, 64)
-	math.ReadBits(Bx, point[:32])
-	math.ReadBits(By, point[32:])
+
+	blob := Bx.Bytes()
+	copy(point[32-len(blob):], blob)
+	blob = By.Bytes()
+	copy(point[64-len(blob):], blob)
```

Comment:
This change reintroduces the byte buffer allocations that were carefully removed through ReadBits. Please make a copy of ReadBits in the secp256k1 package. The function used to live in this package before being moved to common/math in 5f7826270c9e87509fd7731ec64953a5e4761de0.

Reply:
Done, PTAL